src/components/RichText.tsx

```diff
@@ -43,215 +29,114 @@ export function RichText({
   interactiveStyle,
   emojiMultiplier = 1.85,
 }: RichTextProps) {
-  const richText = React.useMemo(
-    () =>
-      value instanceof RichTextAPI ? value : new RichTextAPI({text: value}),
-    [value],
-  )
+  const richText = useMemo(
+    () => value instanceof RichTextAPI ? value : new RichTextAPI({ text: value }),
+    [value]
+  );
 
-  const flattenedStyle = flatten(style)
-  const plainStyles = [a.leading_snug, flattenedStyle]
-  const interactiveStyles = [
-    a.leading_snug,
-    a.pointer_events_auto,
-    flatten(interactiveStyle),
-    flattenedStyle,
-  ]
+  const flattenedStyle = flatten(style);
+  const plainStyles = [a.leading_snug, flattenedStyle];
+  const interactiveStyles = [a.leading_snug, a.pointer_events_auto, flatten(interactiveStyle), flattenedStyle];
 
-  const {text, facets} = richText
+  const { text, facets } = richText;
 
   if (!facets?.length) {
-    if (isOnlyEmoji(text)) {
-      const fontSize =
-        (flattenedStyle.fontSize ?? a.text_sm.fontSize) * emojiMultiplier
-      return (
-        <Text
-          selectable={selectable}
-          testID={testID}
-          style={[plainStyles, {fontSize}]}
-          // @ts-ignore web only -prf
-          dataSet={WORD_WRAP}>
-          {text}
-        </Text>
-      )
-    }
+    return renderSimpleText(text, selectable, testID, numberOfLines, plainStyles, emojiMultiplier);
+  }
+
+  const els = buildTextElements(richText, disableLinks, enableTags, authorHandle, interactiveStyles, onLinkPress);
+
+  return (
+    <Text selectable={selectable} testID={testID} style={plainStyles} numberOfLines={numberOfLines}>
+      {els}
+    </Text>
+  );
+}
+
+function renderSimpleText(text, selectable, testID, numberOfLines, plainStyles, emojiMultiplier) {
+  if (isOnlyEmoji(text)) {
+    const fontSize = (flattenedStyle.fontSize ?? a.text_sm.fontSize) * emojiMultiplier;
     return (
-      <Text
-        selectable={selectable}
-        testID={testID}
-        style={plainStyles}
-        numberOfLines={numberOfLines}
-        // @ts-ignore web only -prf
-        dataSet={WORD_WRAP}>
+      <Text selectable={selectable} testID={testID} style={[plainStyles, { fontSize }]} numberOfLines={numberOfLines}>
         {text}
       </Text>
-    )
+    );
   }
+  return (
+    <Text selectable={selectable} testID={testID} style={plainStyles} numberOfLines={numberOfLines}>
+      {text}
+    </Text>
+  );
+}
+
+const DEFAULT_DOMAIN = 'bsky.social';
+const TLD_REGEX = /.\w+$/;  // Regex to check for a valid top-level domain
```

Comment:
`.` is special in regex and matches anything. So this whole regex is, mostly, a match anything regex.